exporter/sentryexporter/sentry_exporter.go

```diff
@@ -127,6 +132,69 @@ func generateTransactions(transactionMap map[sentry.SpanID]*sentry.Event, orphan
 	return transactions
 }
 
+// convertEventstoSentryExceptions creates a set of sentry events from exception events present in spans.
+func convertEventstoSentryExceptions(events pdata.SpanEventSlice, sentrySpan *sentry.Span, eventList *[]*sentry.Event) {
+	for i := 0; i < events.Len(); i++ {
+		event := events.At(i)
+		if event.Name() != "exception" {
+			continue
+		}
+		var exceptionEscaped bool
+		var exceptionMessage, exceptionStacktrace, exceptionType string
+		event.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
+			switch k {
+			case conventions.AttributeExceptionMessage:
+				exceptionMessage = v.StringVal()
+			case conventions.AttributeExceptionStacktrace:
+				exceptionStacktrace = v.StringVal()
+			case conventions.AttributeExceptionType:
+				exceptionType = v.StringVal()
+			case "exception.escaped":
+				exceptionEscaped = v.BoolVal()
+			}
+			return true
+		})
+		if exceptionMessage == "" && exceptionType == "" {
+			// `At least one of the following sets of attributes is required:
+			// - exception.type
+			// - exception.message`
+			continue
+		}
+		sentryEvent := eventFromError(exceptionMessage, exceptionType, exceptionStacktrace, sentrySpan)
+		*eventList = append(*eventList, sentryEvent)
+	}
+}
+
+// eventFromError creates a sentry event from error event in a span
+func eventFromError(errorMessage, errorType, errorStackTrace string, span *sentry.Span) *sentry.Event {
+	event := sentry.NewEvent()
+
+	event.Contexts["trace"] = sentry.TraceContext{
+		TraceID: span.TraceID,
+		SpanID:  span.SpanID,
+		Op:      span.Op,
+		Status:  span.Status,
+	}
+
+	event.Type = errorType
+	event.Message = errorMessage
+	event.Level = "error"
+	event.Exception = append(event.Exception, sentry.Exception{
```

Comment:
Why are we appending? New events should have no existing exception right?

Reply:
Fair point, will be fixing this in the next commit.